File: gamslib/flowshop/flowshop-scalar.py
#  MIP written by GAMS Convert at 12/13/18 11:34:20
#  
#  Equation counts
#      Total        E        G        L        N        X        C        B
#         58       30       28        0        0        0        0        0
#  
#  Variable counts
#                   x        b        i      s1s      s2s       sc       si
#      Total     cont   binary  integer     sos1     sos2    scont     sint
#         73       37       36        0        0        0        0        0
#  FX      0        0        0        0        0        0        0        0
#  
#  Nonzero counts
#      Total    const       NL      DLL
#        272      272        0        0


from pyomo.environ import *

model = m = ConcreteModel()


m.b1 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b2 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b3 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b4 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b5 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b6 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b7 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b8 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b9 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b10 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b11 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b12 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b13 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b14 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b15 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b16 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b17 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b18 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b19 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b20 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b21 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b22 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b23 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b24 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b25 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b26 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b27 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b28 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b29 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b30 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b31 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b32 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b33 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b34 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b35 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b36 = Var(within=Binary,bounds=(0,1),initialize=0)
m.x37 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x38 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x39 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x40 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x41 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x42 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x43 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x44 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x45 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x46 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x47 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x48 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x49 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x50 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x51 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x52 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x53 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x54 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x55 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x56 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x57 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x58 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x59 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x60 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x61 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x62 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x63 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x64 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x65 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x66 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x67 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x68 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x69 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x70 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x71 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x72 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x73 = Var(within=Reals,bounds=(None,None),initialize=0)

m.obj = Objective(expr=m.x73, sense=minimize)

m.c1 = Constraint(expr=   m.b1 + m.b7 + m.b13 + m.b19 + m.b25 + m.b31 == 1)

m.c2 = Constraint(expr=   m.b2 + m.b8 + m.b14 + m.b20 + m.b26 + m.b32 == 1)

m.c3 = Constraint(expr=   m.b3 + m.b9 + m.b15 + m.b21 + m.b27 + m.b33 == 1)

m.c4 = Constraint(expr=   m.b4 + m.b10 + m.b16 + m.b22 + m.b28 + m.b34 == 1)

m.c5 = Constraint(expr=   m.b5 + m.b11 + m.b17 + m.b23 + m.b29 + m.b35 == 1)

m.c6 = Constraint(expr=   m.b6 + m.b12 + m.b18 + m.b24 + m.b30 + m.b36 == 1)

m.c7 = Constraint(expr=   m.b1 + m.b2 + m.b3 + m.b4 + m.b5 + m.b6 == 1)

m.c8 = Constraint(expr=   m.b7 + m.b8 + m.b9 + m.b10 + m.b11 + m.b12 == 1)

m.c9 = Constraint(expr=   m.b13 + m.b14 + m.b15 + m.b16 + m.b17 + m.b18 == 1)

m.c10 = Constraint(expr=   m.b19 + m.b20 + m.b21 + m.b22 + m.b23 + m.b24 == 1)

m.c11 = Constraint(expr=   m.b25 + m.b26 + m.b27 + m.b28 + m.b29 + m.b30 == 1)

m.c12 = Constraint(expr=   m.b31 + m.b32 + m.b33 + m.b34 + m.b35 + m.b36 == 1)

m.c13 = Constraint(expr=   m.x38 - m.x55 >= 0)

m.c14 = Constraint(expr=   m.x39 - m.x56 >= 0)

m.c15 = Constraint(expr=   m.x40 - m.x57 >= 0)

m.c16 = Constraint(expr=   m.x41 - m.x58 >= 0)

m.c17 = Constraint(expr=   m.x42 - m.x59 >= 0)

m.c18 = Constraint(expr=   m.x44 - m.x61 >= 0)

m.c19 = Constraint(expr=   m.x45 - m.x62 >= 0)

m.c20 = Constraint(expr=   m.x46 - m.x63 >= 0)

m.c21 = Constraint(expr=   m.x47 - m.x64 >= 0)

m.c22 = Constraint(expr=   m.x48 - m.x65 >= 0)

m.c23 = Constraint(expr=   m.x50 - m.x67 >= 0)

m.c24 = Constraint(expr=   m.x51 - m.x68 >= 0)

m.c25 = Constraint(expr=   m.x52 - m.x69 >= 0)

m.c26 = Constraint(expr=   m.x53 - m.x70 >= 0)

m.c27 = Constraint(expr=   m.x54 - m.x71 >= 0)

m.c28 = Constraint(expr=   m.x43 - m.x55 >= 0)

m.c29 = Constraint(expr=   m.x44 - m.x56 >= 0)

m.c30 = Constraint(expr=   m.x45 - m.x57 >= 0)

m.c31 = Constraint(expr=   m.x46 - m.x58 >= 0)

m.c32 = Constraint(expr=   m.x47 - m.x59 >= 0)

m.c33 = Constraint(expr=   m.x48 - m.x60 >= 0)

m.c34 = Constraint(expr=   m.x49 - m.x61 >= 0)

m.c35 = Constraint(expr=   m.x50 - m.x62 >= 0)

m.c36 = Constraint(expr=   m.x51 - m.x63 >= 0)

m.c37 = Constraint(expr=   m.x52 - m.x64 >= 0)

m.c38 = Constraint(expr=   m.x53 - m.x65 >= 0)

m.c39 = Constraint(expr=   m.x54 - m.x66 >= 0)

m.c40 = Constraint(expr= - 3*m.b1 - 6*m.b7 - 3*m.b13 - 5*m.b19 - 5*m.b25 - 7*m.b31 - m.x37 + m.x55 == 0)

m.c41 = Constraint(expr= - 3*m.b2 - 6*m.b8 - 3*m.b14 - 5*m.b20 - 5*m.b26 - 7*m.b32 - m.x38 + m.x56 == 0)

m.c42 = Constraint(expr= - 3*m.b3 - 6*m.b9 - 3*m.b15 - 5*m.b21 - 5*m.b27 - 7*m.b33 - m.x39 + m.x57 == 0)

m.c43 = Constraint(expr= - 3*m.b4 - 6*m.b10 - 3*m.b16 - 5*m.b22 - 5*m.b28 - 7*m.b34 - m.x40 + m.x58 == 0)

m.c44 = Constraint(expr= - 3*m.b5 - 6*m.b11 - 3*m.b17 - 5*m.b23 - 5*m.b29 - 7*m.b35 - m.x41 + m.x59 == 0)

m.c45 = Constraint(expr= - 3*m.b6 - 6*m.b12 - 3*m.b18 - 5*m.b24 - 5*m.b30 - 7*m.b36 - m.x42 + m.x60 == 0)

m.c46 = Constraint(expr= - 5*m.b1 - 4*m.b7 - 2*m.b13 - 4*m.b19 - 4*m.b25 - 5*m.b31 - m.x43 + m.x61 == 0)

m.c47 = Constraint(expr= - 5*m.b2 - 4*m.b8 - 2*m.b14 - 4*m.b20 - 4*m.b26 - 5*m.b32 - m.x44 + m.x62 == 0)

m.c48 = Constraint(expr= - 5*m.b3 - 4*m.b9 - 2*m.b15 - 4*m.b21 - 4*m.b27 - 5*m.b33 - m.x45 + m.x63 == 0)

m.c49 = Constraint(expr= - 5*m.b4 - 4*m.b10 - 2*m.b16 - 4*m.b22 - 4*m.b28 - 5*m.b34 - m.x46 + m.x64 == 0)

m.c50 = Constraint(expr= - 5*m.b5 - 4*m.b11 - 2*m.b17 - 4*m.b23 - 4*m.b29 - 5*m.b35 - m.x47 + m.x65 == 0)

m.c51 = Constraint(expr= - 5*m.b6 - 4*m.b12 - 2*m.b18 - 4*m.b24 - 4*m.b30 - 5*m.b36 - m.x48 + m.x66 == 0)

m.c52 = Constraint(expr= - 5*m.b1 - 2*m.b7 - 4*m.b13 - 6*m.b19 - 3*m.b25 - 6*m.b31 - m.x49 + m.x67 == 0)

m.c53 = Constraint(expr= - 5*m.b2 - 2*m.b8 - 4*m.b14 - 6*m.b20 - 3*m.b26 - 6*m.b32 - m.x50 + m.x68 == 0)

m.c54 = Constraint(expr= - 5*m.b3 - 2*m.b9 - 4*m.b15 - 6*m.b21 - 3*m.b27 - 6*m.b33 - m.x51 + m.x69 == 0)

m.c55 = Constraint(expr= - 5*m.b4 - 2*m.b10 - 4*m.b16 - 6*m.b22 - 3*m.b28 - 6*m.b34 - m.x52 + m.x70 == 0)

m.c56 = Constraint(expr= - 5*m.b5 - 2*m.b11 - 4*m.b17 - 6*m.b23 - 3*m.b29 - 6*m.b35 - m.x53 + m.x71 == 0)

m.c57 = Constraint(expr= - 5*m.b6 - 2*m.b12 - 4*m.b18 - 6*m.b24 - 3*m.b30 - 6*m.b36 - m.x54 + m.x72 == 0)

m.c58 = Constraint(expr= - m.x72 + m.x73 >= 0)

File: gamslib/flowshop/flowshop-scalar.jl
#  MIP written by GAMS Convert at 10/11/20 13:51:32
#  
#  Equation counts
#      Total        E        G        L        N        X        C        B
#         58       30       28        0        0        0        0        0
#  
#  Variable counts
#                   x        b        i      s1s      s2s       sc       si
#      Total     cont   binary  integer     sos1     sos2    scont     sint
#         73       37       36        0        0        0        0        0
#  FX      0        0        0        0        0        0        0        0
#  
#  Nonzero counts
#      Total    const       NL      DLL
#        272      272        0        0


using JuMP
using MathOptInterface
const MOI = MathOptInterface

model = m = Model()

@variable(m, 0 <= b1 <= 1, binary=true, start=0)
@variable(m, 0 <= b2 <= 1, binary=true, start=0)
@variable(m, 0 <= b3 <= 1, binary=true, start=0)
@variable(m, 0 <= b4 <= 1, binary=true, start=0)
@variable(m, 0 <= b5 <= 1, binary=true, start=0)
@variable(m, 0 <= b6 <= 1, binary=true, start=0)
@variable(m, 0 <= b7 <= 1, binary=true, start=0)
@variable(m, 0 <= b8 <= 1, binary=true, start=0)
@variable(m, 0 <= b9 <= 1, binary=true, start=0)
@variable(m, 0 <= b10 <= 1, binary=true, start=0)
@variable(m, 0 <= b11 <= 1, binary=true, start=0)
@variable(m, 0 <= b12 <= 1, binary=true, start=0)
@variable(m, 0 <= b13 <= 1, binary=true, start=0)
@variable(m, 0 <= b14 <= 1, binary=true, start=0)
@variable(m, 0 <= b15 <= 1, binary=true, start=0)
@variable(m, 0 <= b16 <= 1, binary=true, start=0)
@variable(m, 0 <= b17 <= 1, binary=true, start=0)
@variable(m, 0 <= b18 <= 1, binary=true, start=0)
@variable(m, 0 <= b19 <= 1, binary=true, start=0)
@variable(m, 0 <= b20 <= 1, binary=true, start=0)
@variable(m, 0 <= b21 <= 1, binary=true, start=0)
@variable(m, 0 <= b22 <= 1, binary=true, start=0)
@variable(m, 0 <= b23 <= 1, binary=true, start=0)
@variable(m, 0 <= b24 <= 1, binary=true, start=0)
@variable(m, 0 <= b25 <= 1, binary=true, start=0)
@variable(m, 0 <= b26 <= 1, binary=true, start=0)
@variable(m, 0 <= b27 <= 1, binary=true, start=0)
@variable(m, 0 <= b28 <= 1, binary=true, start=0)
@variable(m, 0 <= b29 <= 1, binary=true, start=0)
@variable(m, 0 <= b30 <= 1, binary=true, start=0)
@variable(m, 0 <= b31 <= 1, binary=true, start=0)
@variable(m, 0 <= b32 <= 1, binary=true, start=0)
@variable(m, 0 <= b33 <= 1, binary=true, start=0)
@variable(m, 0 <= b34 <= 1, binary=true, start=0)
@variable(m, 0 <= b35 <= 1, binary=true, start=0)
@variable(m, 0 <= b36 <= 1, binary=true, start=0)
@variable(m, 0 <= x37, start=0)
@variable(m, 0 <= x38, start=0)
@variable(m, 0 <= x39, start=0)
@variable(m, 0 <= x40, start=0)
@variable(m, 0 <= x41, start=0)
@variable(m, 0 <= x42, start=0)
@variable(m, 0 <= x43, start=0)
@variable(m, 0 <= x44, start=0)
@variable(m, 0 <= x45, start=0)
@variable(m, 0 <= x46, start=0)
@variable(m, 0 <= x47, start=0)
@variable(m, 0 <= x48, start=0)
@variable(m, 0 <= x49, start=0)
@variable(m, 0 <= x50, start=0)
@variable(m, 0 <= x51, start=0)
@variable(m, 0 <= x52, start=0)
@variable(m, 0 <= x53, start=0)
@variable(m, 0 <= x54, start=0)
@variable(m, 0 <= x55, start=0)
@variable(m, 0 <= x56, start=0)
@variable(m, 0 <= x57, start=0)
@variable(m, 0 <= x58, start=0)
@variable(m, 0 <= x59, start=0)
@variable(m, 0 <= x60, start=0)
@variable(m, 0 <= x61, start=0)
@variable(m, 0 <= x62, start=0)
@variable(m, 0 <= x63, start=0)
@variable(m, 0 <= x64, start=0)
@variable(m, 0 <= x65, start=0)
@variable(m, 0 <= x66, start=0)
@variable(m, 0 <= x67, start=0)
@variable(m, 0 <= x68, start=0)
@variable(m, 0 <= x69, start=0)
@variable(m, 0 <= x70, start=0)
@variable(m, 0 <= x71, start=0)
@variable(m, 0 <= x72, start=0)
@variable(m, x73, start=0)

@objective(m, Min, x73)

@constraint(m, b1 + b7 + b13 + b19 + b25 + b31 == 1)

@constraint(m, b2 + b8 + b14 + b20 + b26 + b32 == 1)

@constraint(m, b3 + b9 + b15 + b21 + b27 + b33 == 1)

@constraint(m, b4 + b10 + b16 + b22 + b28 + b34 == 1)

@constraint(m, b5 + b11 + b17 + b23 + b29 + b35 == 1)

@constraint(m, b6 + b12 + b18 + b24 + b30 + b36 == 1)

@constraint(m, b1 + b2 + b3 + b4 + b5 + b6 == 1)

@constraint(m, b7 + b8 + b9 + b10 + b11 + b12 == 1)

@constraint(m, b13 + b14 + b15 + b16 + b17 + b18 == 1)

@constraint(m, b19 + b20 + b21 + b22 + b23 + b24 == 1)

@constraint(m, b25 + b26 + b27 + b28 + b29 + b30 == 1)

@constraint(m, b31 + b32 + b33 + b34 + b35 + b36 == 1)

@constraint(m, x38 - x55 >= 0)

@constraint(m, x39 - x56 >= 0)

@constraint(m, x40 - x57 >= 0)

@constraint(m, x41 - x58 >= 0)

@constraint(m, x42 - x59 >= 0)

@constraint(m, x44 - x61 >= 0)

@constraint(m, x45 - x62 >= 0)

@constraint(m, x46 - x63 >= 0)

@constraint(m, x47 - x64 >= 0)

@constraint(m, x48 - x65 >= 0)

@constraint(m, x50 - x67 >= 0)

@constraint(m, x51 - x68 >= 0)

@constraint(m, x52 - x69 >= 0)

@constraint(m, x53 - x70 >= 0)

@constraint(m, x54 - x71 >= 0)

@constraint(m, x43 - x55 >= 0)

@constraint(m, x44 - x56 >= 0)

@constraint(m, x45 - x57 >= 0)

@constraint(m, x46 - x58 >= 0)

@constraint(m, x47 - x59 >= 0)

@constraint(m, x48 - x60 >= 0)

@constraint(m, x49 - x61 >= 0)

@constraint(m, x50 - x62 >= 0)

@constraint(m, x51 - x63 >= 0)

@constraint(m, x52 - x64 >= 0)

@constraint(m, x53 - x65 >= 0)

@constraint(m, x54 - x66 >= 0)

@constraint(m,  - 3*b1 - 6*b7 - 3*b13 - 5*b19 - 5*b25 - 7*b31 - x37 + x55 == 0)

@constraint(m,  - 3*b2 - 6*b8 - 3*b14 - 5*b20 - 5*b26 - 7*b32 - x38 + x56 == 0)

@constraint(m,  - 3*b3 - 6*b9 - 3*b15 - 5*b21 - 5*b27 - 7*b33 - x39 + x57 == 0)

@constraint(m,  - 3*b4 - 6*b10 - 3*b16 - 5*b22 - 5*b28 - 7*b34 - x40 + x58 == 0)

@constraint(m,  - 3*b5 - 6*b11 - 3*b17 - 5*b23 - 5*b29 - 7*b35 - x41 + x59 == 0)

@constraint(m,  - 3*b6 - 6*b12 - 3*b18 - 5*b24 - 5*b30 - 7*b36 - x42 + x60 == 0)

@constraint(m,  - 5*b1 - 4*b7 - 2*b13 - 4*b19 - 4*b25 - 5*b31 - x43 + x61 == 0)

@constraint(m,  - 5*b2 - 4*b8 - 2*b14 - 4*b20 - 4*b26 - 5*b32 - x44 + x62 == 0)

@constraint(m,  - 5*b3 - 4*b9 - 2*b15 - 4*b21 - 4*b27 - 5*b33 - x45 + x63 == 0)

@constraint(m,  - 5*b4 - 4*b10 - 2*b16 - 4*b22 - 4*b28 - 5*b34 - x46 + x64 == 0)

@constraint(m,  - 5*b5 - 4*b11 - 2*b17 - 4*b23 - 4*b29 - 5*b35 - x47 + x65 == 0)

@constraint(m,  - 5*b6 - 4*b12 - 2*b18 - 4*b24 - 4*b30 - 5*b36 - x48 + x66 == 0)

@constraint(m,  - 5*b1 - 2*b7 - 4*b13 - 6*b19 - 3*b25 - 6*b31 - x49 + x67 == 0)

@constraint(m,  - 5*b2 - 2*b8 - 4*b14 - 6*b20 - 3*b26 - 6*b32 - x50 + x68 == 0)

@constraint(m,  - 5*b3 - 2*b9 - 4*b15 - 6*b21 - 3*b27 - 6*b33 - x51 + x69 == 0)

@constraint(m,  - 5*b4 - 2*b10 - 4*b16 - 6*b22 - 3*b28 - 6*b34 - x52 + x70 == 0)

@constraint(m,  - 5*b5 - 2*b11 - 4*b17 - 6*b23 - 3*b29 - 6*b35 - x53 + x71 == 0)

@constraint(m,  - 5*b6 - 2*b12 - 4*b18 - 6*b24 - 3*b30 - 6*b36 - x54 + x72 == 0)

@constraint(m,  - x72 + x73 >= 0)
